core/types/subtyping.cc

```diff
@@ -1578,8 +1578,34 @@ bool Types::isSubTypeUnderConstraint(const GlobalState &gs, TypeConstraint &cons
     // Note: order of cases here matters! We can't lose "and" information in t1 early and we can't
     // lose "or" information in t2 early.
     if (auto o1 = cast_type<OrType>(t1)) { // 7, 8, 9
-        return Types::isSubTypeUnderConstraint(gs, constr, o1->left, t2, mode, errorDetailsCollector) &&
-               Types::isSubTypeUnderConstraint(gs, constr, o1->right, t2, mode, errorDetailsCollector);
+        auto subCollectorLeft = errorDetailsCollector.newCollector();
+        auto isSubTypeOfLeft = Types::isSubTypeUnderConstraint(gs, constr, o1->left, t2, mode, errorDetailsCollector);
+        if (!isSubTypeOfLeft) {
+            if constexpr (shouldAddErrorDetails) {
+                // This if is to handle the T.nilable(X) < X case; it's not useful say that nil is not a subtype of X
+                if (!o1->left.isNilClass()) {
+                    auto message = ErrorColors::format("`{}` (the left side of the `{}`) is not a subtype of `{}`",
+                                                       o1->left.show(gs), "T.any", t2.show(gs));
```

Comment:
I wonder whether it's attempting to sneak into the error message here something like, "... and the right side of the T.any wasn't even checked because the left side failed"

If there's a short way to say that I wonder if it would be useful. Because I could see something like

```
T.any(A1, A2) <: Unrelated
```

you're going to get something like `A1` is not a subtype of `Unrelated`, but not anything about `A2`, and I could see people misinterpreting that as `A2` being a subtype of `Unrelated`